library/Dotfiles.re

```diff
@@ -0,0 +1,33 @@
+exception Version_Not_Provided;
+
+let versionString = fileName => {
+  try%lwt (
+    Lwt_io.lines_of_file(fileName)
+    |> Lwt_stream.to_list
+    |> Lwt.map(List.hd)
+    |> Lwt.map(String.trim)
+    |> Lwt.map(versionString => Some(versionString))
+  ) {
+  | Unix.Unix_error(Unix.ENOENT, _, _) => Lwt.return(None)
+  };
+};
+
+let getVersion = () => {
+  let%lwt cwd = Lwt_unix.getcwd();
+  let%lwt nodeVersion = versionString(Filename.concat(cwd, ".node-version"))
+  and nvmrc = versionString(Filename.concat(cwd, ".nvmrc"));
+
+  switch (nodeVersion, nvmrc) {
+  | (None, None) => Lwt.fail(Version_Not_Provided)
+  | (Some(nodeVersionString), Some(nvmrcString)) =>
+    if (nodeVersionString === nvmrcString) {
```

Comment:
This can even be reduced further using:

```reason 
  | (Some(v1), Some(v2) when v1 == v2 =>
```
Then, no nesting in our pattern matching! Wdyt?

Reply:
Or even better; check if it isn’t equal first :)